app/services/doc_auth/lexis_nexis/request.rb

```diff
@@ -0,0 +1,153 @@
+module DocAuth
+  module LexisNexis
+    class Request
+      def fetch
+        http_response = send_http_request
+        return handle_invalid_response(http_response) unless http_response.success?
+
+        handle_http_response(http_response)
+      rescue Faraday::ConnectionFailed, Faraday::TimeoutError => e
+        handle_connection_error(e)
+      end
+
+      private
+
+      def send_http_request
+        case method.downcase.to_sym
+        when :post
+          send_http_post_request
+        when :get
+          send_http_get_request
+        end
+      end
+
+      def handle_http_response(_response)
+        raise NotImplementedError
+      end
+
+      def handle_invalid_response(http_response)
+        message = [
+          self.class.name,
+          'Unexpected HTTP response',
+          http_response.status,
+        ].join(' ')
+        exception = RuntimeError.new(message)
+        NewRelic::Agent.notice_error(exception)
+        DocAuth::Response.new(
+          success: false,
+          errors: { network: I18n.t('errors.doc_auth.lexisnexis_network_error') },
+          exception: exception,
+        )
+      end
+
+      def handle_connection_error(exception)
+        NewRelic::Agent.notice_error(exception)
+        DocAuth::Response.new(
+          success: false,
+          errors: { network: I18n.t('errors.doc_auth.lexisnexis_network_error') },
+          exception: exception,
+        )
+      end
+
+      def send_http_get_request
+        faraday_connection.get
+      end
+
+      def send_http_post_request
+        faraday_connection.post do |req|
+          req.body = body
+        end
+      end
+
+      def faraday_connection
+        retry_options = {
+          max: 2,
+          interval: 0.05,
+          interval_randomness: 0.5,
+          backoff_factor: 2,
+          retry_statuses: [404, 500],
+          retry_block: lambda do |_env, _options, retries, exc|
+            NewRelic::Agent.notice_error(exc, custom_params: { retry: retries })
+          end,
+        }
+
+        Faraday.new(request: faraday_request_params, url: url.to_s, headers: headers) do |conn|
+          conn.request :retry, retry_options
+          conn.adapter :net_http
+        end
+      end
+
+      def faraday_request_params
+        timeout = Figaro.env.lexisnexis_timeout&.to_i || 45
+        { open_timeout: timeout, timeout: timeout }
+      end
+
+      def path
+        "/restws/identity/v3/#{account_id}/workflows/#{workflow}/conversation"
+      end
+
+      def method
+        :get
+      end
+
+      def url
+        URI.join(Figaro.env.lexisnexis_base_url, path)
+      end
+
+      def headers
+        {
+          Authorization: "Basic #{encoded_credentials}",
```

Comment:
Faraday has `#basic_auth` which might make this easier? https://www.rubydoc.info/github/lostisland/faraday/Faraday%2FConnection:basic_auth

Reply:
Good catch!